R/data-structure.R

```diff
@@ -5,30 +5,46 @@
 #'
 #' To me, partial matching is often more annoying and surprising than
 #' convenient. It can lead to bugs that are very hard to discover, and I have
-#' been bitten for many times. When I write \code{x$name}, I always mean
-#' precisely \code{name}. You should use a modern code editor to autocomplate
-#' the \code{name} if it is too long to type, instead of using a partial name.
+#' been bitten by it many times. When I write \code{x$name}, I always mean
+#' precisely \code{name}. You should use a modern code editor to autocomplete
+#' the \code{name} if it is too long to type, instead of using partial names.
 #' @param ... Objects (list elements), possibly named. Ignored in the
 #'   \code{print()} method.
 #' @export
-#' @return \code{strict_list()} returns a list with the class
-#'   \code{xfun_strict_list}.
+#' @return \code{strict_list()} and \code{as_strict_list()} both return a list
+#'   with the class \code{xfun_strict_list}. Whereas \code{as_strict_list()}
+#'   attempts to coerce its argument \code{x} to a list if necessary,
+#'   \code{xfun_strict_list} just wraps its argument \code{...} in a list,
+#'   i.e. it will add another list level regardless if \code{...} already is
+#'   of type list.
 #' @examples library(xfun)
-#' (z = strict_list(aaa = 'I am aaa', b = 1:5))
+#' z <- strict_list(aaa = 'I am aaa', b = 1:5)
```

Comment:
Please refrain from making cosmetic changes: https://yihui.name/en/2018/11/cosmetic-changes/ (I'm sorry, but please use `=` here for consistency with the rest of code in this package).

BTW, the outer parentheses were there for a reason (printing): https://yihui.name/en/2017/06/top-level-r-expressions/

Reply:
> Please refrain from making cosmetic changes (...)

Thanks for the link, useful read! I guess you're right. (And I actually didn't realize that there are more ` =  ` assignments left in the file after my changes. So I updated my PR to use ` = ` everywhere :)

But I have to remind you that you actually do use the `<-` operator in this package, e.g. in `encoding.R`, `utils.R` or `revcheck.R`... 😜

> BTW, the outer parentheses were there for a reason (printing) (...)

Thanks for this link as well! I've read it now and didn't know about this 🙈:

> You can force a top-level expression to be visible (hence printed) by wrapping it in `()`

I changed the parentheses to how they were.

Reply:
Thanks for the reminder about the use of `<-`! I used them deliberately in these places for a reason. I definitely don't avoid `<-` unconditionally. It is good for what it is good for.

John Mount has explained it well here: http://www.win-vector.com/blog/2016/12/the-case-for-using-in-r/

Reply:
Thanks for this link, too! I totally get why you generally prefer `=` over `<-` for assignments.

(Personally I just stick to [Hadyley's style guide](http://adv-r.had.co.nz/Style.html) and use `<-` since the benefits of using `=` don't convince me enough to deviate from common standards/practices; BTW: If you don't already, I'd suggest to use the font [FiraCode](https://github.com/tonsky/FiraCode) for programming (instructions for RStudio [here](https://github.com/tonsky/FiraCode/wiki/RStudio-instructions)) which does increase readability a lot and at least helps you spot (admittedly pretty artificial) syntax lapses like an unwanted space in between `< -`.)

Now I'm wondering: When you do something like

``` r
if ( (bla <- 2-2) == 0 ) print(paste0("yup, bla is: ", bla))
#> [1] "yup, bla is: 0"
```

you still _deliberately_ use `<-` instead of `=`, although the latter would produce the same result:

``` r
if ( (bla = 2-2) == 0 ) print(paste0("yup, bla is: ", bla))
#> [1] "yup, bla is: 0"
```

What's your motivation behind this? To make sure you don't accidentally confuse the single `=` with the comparator `==`?

Reply:
Ok, makes sense. Thanks for the explanation!